### PR TITLE
Implemented JSONString

### PR DESCRIPTION
**Pull request by okinealb**

### CODE REVIEW

File: JSONString.java [hunk 1]
```diff
@@ -58,7 +64,7 @@ public int hashCode() {
    * Write the value as JSON.
    */
   public void writeJSON(PrintWriter pen) {
-                        // STUB
+    pen.print(this.toString());
```
Comment (eoyoa) [inline JSONString.java:67]: Don't forget to flush the `PrintWriter`, it is not guaranteed auto-flush.

File: JSONString.java [hunk 2]
```diff
@@ -33,21 +33,27 @@ public JSONString(String value) {
    * Convert to a string (e.g., for printing).
    */
   public String toString() {
-    return "";          // STUB
+    return (this.value == null)
```
Comment (eoyoa) [inline JSONString.java:36]: Is `JSONString` nullable? What's the point of JSONConstant then?

File: JSONString.java [hunk 3]
```diff
@@ -33,21 +33,27 @@ public JSONString(String value) {
...
+            || ((other instanceof JSONString)
+                && (this.value.equals(((JSONString) other).value))));
   } // equals(Object)
 
   /**
    * Compute the hash code.
    */
   public int hashCode() {
-    return 0;           // STUB
+    return (this.value == null)
```
Comment (eoyoa) [inline JSONString.java:54]: Again, is `JSONString` nullable?

**Review by eoyoa — CHANGES_REQUESTED** [review]

File: JSONString.java (hunk 2 above)
Comment (okinealb) [reply JSONString.java:36]: No lol. For some reason, Rebelsky did that in JSONConstant (which was already implemented) so I mirrored that for consistency

File: JSONString.java (hunk 3 above)
Comment (okinealb) [reply JSONString.java:54]: Like above, it'll never be null, but its consistent

**Pull request closed without merging**